sgeop/geometry.py

```diff
@@ -354,3 +371,36 @@ def snap_to_targets(edgelines, poly, snap_to, secondary_snap_to=None):
                     stacklevel=2,
                 )
     return to_add, to_split
+
+
+def _prep_components(lines: np.ndarray) -> tuple[pd.Series, pd.Series, gpd.GeoSeries]:
+    """Helper for preparing graph components & labels in PySAL."""
+
+    # cast edgelines to gdf
+    lines = gpd.GeoDataFrame(geometry=lines)
+
+    # build queen contiguity on edgelines and extract component labels
+    not_empty = ~lines.is_empty
+    not_nan = ~lines.geometry.isna()
+    lines = lines[not_empty | not_nan]
+    comp_labels = graph.Graph.build_contiguity(lines, rook=False).component_labels
+
+    # compute size of each component
+    comp_counts = comp_labels.value_counts()
+
+    # get MultiLineString geometry per connected component
+    components = lines.dissolve(comp_labels)
+
+    return comp_labels, comp_counts, components
+
+
+def _split_add(line: shapely.LineString, splits: list, adds: list) -> tuple[list]:
+    """Helper for preparing splitter points & added lines."""
+    splits.append(_last_point(line))
+    adds.append(line)
+    return splits, adds
+
+
+def _last_point(line: shapely.LineString) -> shapely.Point:
+    """Helper for returning final point on a line."""
+    return shapely.get_point(line, -1)
```

Comment:
I'd probably prefer using `shapely.get_point(line, -1)` and it is explicit

Reply:
I've got no problem with that. I'll dump that helper and remove from testing.